src/commands/output.rs

```diff
@@ -25,6 +27,65 @@ pub struct OutputFormatter {
 }
 
 impl OutputFormatter {
+    pub fn format_output(&self, response: MonitorResponse, status: TrackStatus, output_format: &str) -> Output {
```

Comment:
This is somewhat blurring the lines of the purposes of `OutputFormatter`, as it should exclusively format the `Output` struct. Though I'd need to give it some thought on how to make it better. I don't want to block your changes for that.